File: backend/src/main/java/com/powerme/dto/ChargingStationDetailDto.java
package com.powerme.dto;

import com.powerme.enums.SocketType;
import com.powerme.enums.StationStatus;
import java.time.LocalTime;

public record ChargingStationDetailDto(
        Long id,
        String name,
        SocketType socketType,
        double powerKw,
        StationStatus status,
        String imagePath,
        double hourlyRate,
        boolean active,
        LocalTime availableFrom,
        LocalTime availableTo
) {
}

File: frontend/src/app/core/models/dto/charging-station-detail.dto.ts
import { SocketType } from '../enums/socket-type.enum';
import { StationStatus } from '../enums/station-status.enum';

export interface ChargingStationDetailDto {
  id: number;
  name: string;
  socketType: SocketType;
  powerKw: number;
  status: StationStatus;
  imagePath: string | null;
  hourlyRate: number;
  active: boolean;
  /** Plage horaire journalière (optionnelle) */
  availableFrom?: string; // 'HH:mm'
  availableTo?: string; // 'HH:mm'
}
